R/tag_query.R

```diff
@@ -394,109 +394,14 @@ as.character.htmltools.tag.query <- function(x, ...) {
-#' conversion allows for element alterations to happen in place (pass by
-#' reference). Meaning that if a css class is added to each selected tag
-#' environment using `$addClass()` and the result of the method call is ignored,
-#' the selected tag environments within the tag query object will still contain
-#' the class addition.  The added class will exist when the tag query tag
-#' environment are converted back to standard tags objects with `$asTags()`.
-#'
-#' Tag environments also contain an extra field, `.$parent`. The `.$parent`
-#' value contains their parent tag environment. The top level tags supplied to
-#' `tagQuery()` will also have a shared parent element. (The shared parent
-#' element will have a `NULL` `.$parent` value.) This allows for performing
-#' sibling alterations at the top level of the tag query object.
-#'
-#' The set of tag environments in a pointing to each other within a tag query
-#' object can be thought of as a linked list while allowing for a "1 to many"
-#' parent to child relationship and up to 1 parent per child.
-#'
-#' ## Tag Query
-#'
-#' A `tagQuery()` behaves simliar to an R6 object in that internal values are
-#' altered in place. (but a `tagQuery()` object is not implemented with `R6`).
-#' The `tagQuery()`'s methods will return itself as much as possible, unless the
-#' method is directly asking for information, e.g. `$selected()` or `$asTags()`.
-#'
-#' Internally, two important pieces of information are maintained: the root
-#' elements and the selected elements. The root tag environment will always
-#' point (after upgrading to a tag environment) to the original tag object
-#' provided to `tagQuery(tag=)`. However, the selected elements are a list of
-#' tag environments that update for every `$find(cssSelector)` call.  The
-#' selected elements are initialized to a list containing the `root` tag
-#' environment. All `tagQuery()` methods will act on the selected elements
-#' unless declared otherwise.
-#'
-#' Tag query objects can be created from other tag query objects. Note, unless
-#' there is an intermediate call to `$asTags()`, the original and new tag query
-#' objects will share the same tag environments. The new tag query object will
-#' have its selected elements reset. For example:
-#'
-#' ```r
-#' x <- tagQuery(div())
-#' y <- tagQuery(x)
-#' z <- tagQuery(x$asTags(selected = FALSE))
-#'
-#' # Add an example class
-#' y$addClass("example")
-#'
-#' # Show `x` and `y` both have the new class
-#' x$asTags()
-#' #> <div class="example"></div>
-#' y$asTags()
-#' #> <div class="example"></div>
-#'
-#' # `z` is isolated from the changes in `x` and `y` due to the `$asTags()`
-#' z$asTags()
-#' #> <div></div>
-#' ```
-#'
-#'
-#' @section Limitations:
-#'
-#'   `tagQuery()`s can **not** be used directly within typical `tag` locations.
-#'   An error should be thrown. Instead, please call `$asTags()` to retrieve the
-#'   tag structures of the selected tag elements or root element respectively.
```

Comment:
These concepts won't go away completely, they'll just likely be towards the end of the tagQuery article